File: core/src/main/java/io/nirahtech/erp/core/Siret.java
package io.nirahtech.erp.core;

public final class Siret {
    private final long code;
    private static final int CODE_LENGTH = 14;

    private Siret(long code) {
        this.code = code;
    }

    public static Siret create(long code) throws IllegalArgumentException {
        if (String.valueOf(code).length() != CODE_LENGTH) {
            throw new IllegalArgumentException(String.format("SIRET must be a number with %s characters.", CODE_LENGTH));
        }
        return new Siret(code);
    }

    public long getCode() {
        return code;
    }

}

File: webapp/src/main/react/src/data/Siret.ts
class Siret {
    private readonly code: number;
    private static readonly CODE_LENGTH: number = 14;
  
    private constructor(code: number) {
      this.code = code;
    }
  
    static create(code: number): Siret {
      if (String(code).length !== Siret.CODE_LENGTH) {
        throw new Error(`SIRET must be a number with ${14} characters.`);
      }
      return new Siret(code);
    }
  
    getCode(): number {
      return this.code;
    }
  }
export default Siret;  